tests/integration/tests/charm_tests/openstack_exporter.py

```diff
@@ -0,0 +1,304 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical
+#
+# See LICENSE file for licensing details.
+
+import json
+import logging
+import subprocess
+import tempfile
+import unittest
+from pathlib import Path
+
+import yaml
+from zaza import model
+
+from charm import CLOUD_NAME, OS_CLIENT_CONFIG, SNAP_NAME, UPSTREAM_SNAP
+
+logger = logging.getLogger(__name__)
+
+APP_NAME = "openstack-exporter"
+STATUS_TIMEOUT = 120  # must be larger than the time defined in zaza-yaml
+
+VERSION_COLUMN = 1
+REVISION_COLUMN = 2
+CHANNEL_COLUMN = 3
+
+
+class OpenstackExporterBaseTest(unittest.TestCase):
```

Comment:
The Charm style guide asks using `pytest-operator` using `pytest`: https://juju.is/docs/sdk/styleguide#heading--functional-tests, is that applicable here?

Reply:
I think it is applicable, but this charm uses zaza to manage integration/functional tests. Switching to pytest-operator would be a large effort here, as it would involve rewriting all the tests. I'm not sure why we went with zaza rather than pytest-operator actually.

Reply:
@tonyandrewmeyer any thoughts here please?

Reply:
Charm-tech is intending to develop a new integration test tool next cycle to replace using pytest-operator/python-libjuju, strongly based on the Juju CLI ([OP050](https://docs.google.com/document/d/1sBt8E6GyJH39IjgAtgoe6CLtmFybN3p2XImSS95SsC4/edit), although that's very much still being developed).

I'd definitely encourage moving away from Zaza, but this would be a bad time to do that, rather than a few months from now.

Reply:
The reason for using zaza is because we need to deploy whole Charm Openstack for testing.

The effort to change that will be re-implement the configure function `zaza.openstack.charm_tests.setup.setup_export_ssl_ca_config`, changing our CI pipeline, and somehow re-implement our integration test. It's not a small change from my POV.

Reply:
Yes we can close this issue